app/models/podcast_import.rb

```diff
@@ -0,0 +1,267 @@
+      end
+    end
+
+    { name: name, email: email }
+  end
+
+  def parse_itunes_categories(feed)
+    itunes_cats = {}
+    Array(feed.itunes_categories).map(&:strip).select { |c| !c.blank? }.each do |cat|
+      if ITunesCategoryValidator.category?(cat)
+        itunes_cats[cat] ||= []
+      elsif parent_cat = ITunesCategoryValidator.subcategory?(cat)
+        itunes_cats[parent_cat] ||= []
+        itunes_cats[parent_cat] << cat
+      end
+    end
+
+    itunes_cats.keys.map { |n| { name: n, subcategories: itunes_cats[n] } }
+  end
+
+  def parse_categories(feed)
+    mcat = Array(feed.media_categories).map(&:strip)
+    rcat = Array(feed.categories).map(&:strip)
+    (mcat + rcat).compact.uniq
+  end
+
+  def parse_keywords(feed)
+    ikey = Array(feed.itunes_keywords).map(&:strip)
+    mkey = Array(feed.media_keywords).map(&:strip)
+    (ikey + mkey).compact.uniq
+  end
+
+  def create_stories
+    feed.entries.map do |entry|
+      story = create_story(entry, series)
+      create_episode(story, entry)
+      story
+    end
+  end
+
+  def create_story(entry, series)
+    story = series.stories.create!(
+      title: entry[:title],
+      short_description: entry[:itunes_subtitle],
+      description: entry[:description],
+      tags: entry[:categories],
+      published_at: entry[:published]
+    )
+
+    # add the audio version
+    version = story.audio_versions.create!(
+      audio_version_template: template,
+      label: 'Podcast Audio',
+      explicit: entry[:itunes_explicit]
+    )
+
+    # add the audio
+    enclosure = enclosure_url(entry)
+    version.audio_files.create!(label: 'Segment A', upload: enclosure) if enclosure
+
+    # add the image
+    story.images.create!(upload: entry.itunes_image) if entry.itunes_image
+
+    story
+  end
+
+  def enclosure_url(entry)
+    entry[:feedburner_orig_enclosure_link] || entry[:enclosure].try(:url)
+  end
+
+  def create_episode(story, entry)
+    # create the distro from the story
+    distro = StoryDistributions::EpisodeDistribution.create(
+      distribution: distribution,
+      story: story,
+      guid: entry.entry_id
+    )
+
+    create_attributes = {}
+    create_attributes[:author] = person(entry[:itunes_author] || entry[:author] || entry[:creator])
+    create_attributes[:block] = (entry[:itunes_block] == 'yes')
+    create_attributes[:explicit] = entry[:itunes_explicit]
+    create_attributes[:guid] = entry.entry_id
+    create_attributes[:is_closed_captioned] = (entry[:itunes_is_closed_captioned] == 'yes')
+    create_attributes[:is_perma_link] = entry[:is_perma_link]
+    create_attributes[:keywords] = (entry[:itunes_keywords] || '').split(',').map(&:strip)
+    create_attributes[:position] = entry[:itunes_order]
+    create_attributes[:url] = episode_url(entry)
+
+    distro.add_episode_to_feeder(create_attributes)
+  end
+
+  def episode_url(entry)
+    url = entry[:feedburner_orig_link] || entry[:url] || entry[:link]
+    if url =~ /libsyn\.com/
```

Comment:
libsyn sets the `link` to the same value as the mp3 enclosure on libsyn.
We don't want to keep linking to libsyn after importing the feed, so I update the url in that special case.